pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/FilterParser.java

```diff
@@ -318,17 +404,17 @@ private Long parseNumber() throws Exception {
     private String parseDigits() throws Exception {
         String result;
         int i = index;
-        int filterLength = filterString.length();
+        int filterLength = filterByteArr.length;
 
         // allow sign
         if (filterLength > 0) {
-            int chr = filterString.charAt(i);
+            int chr = (char) filterByteArr[i];
```

Comment:
if we cast to (char), should the datatype be char as well, not int ? i know it works either way, but for consistency...